inspector/src/components/actionTabs/tabs/performanceViewer/performanceViewerComponent.tsx

```diff
@@ -129,52 +133,36 @@ export const PerformanceViewerComponent: React.FC<IPerformanceViewerComponentPro
     };
 
     const onToggleRecording = () => {
-        if (recordingState === RecordingState.Recording) {
-            setRecordingState(RecordingState.NotRecording);
+        if (!performanceCollector?.isStarted) {
+            performanceCollector?.start(true);
         } else {
-            setRecordingState(RecordingState.Recording);
+            performanceCollector?.stop();
         }
     };
 
-    useEffect(() => {
-        const perfCollector = scene.getPerfCollector();
-        perfCollector.addCollectionStrategies(...defaultStrategies);
+    const addStrategies = (perfCollector : PerformanceViewerCollector) => {
+        perfCollector.addCollectionStrategies(...defaultStrategiesList);
         if (ComputePressureObserverWrapper.IsAvailable) {
             perfCollector.addCollectionStrategies({
                 strategyCallback: PerfCollectionStrategy.CpuStrategy(), category: IPerfMetadataCategory.FrameSteps
             });
         }
-        setPerformanceCollector(perfCollector);
-    }, []);
-
-    useEffect(() => {
-        if (isOpen && !isLoaded) {
-            setRecordingState(RecordingState.Recording);
-        } else {
-            setRecordingState(RecordingState.NotRecording);
-        }
-    }, [isOpen]);
+    }
 
     useEffect(() => {
-        if (recordingState === RecordingState.Recording) {
-            performanceCollector?.start(true); //preserve data
-        } else {
-            performanceCollector?.stop();
-        }
-
-        return () => {
-            performanceCollector?.stop();
-        };
-    }, [recordingState]);
+        const perfCollector = scene.getPerfCollector();

```

Comment:
any cleanup needed here when the component is disposed?

Reply:
I didn't want to call any stop or clear cleanups because when we open the Popup window and then change tabs, this component is disposed, and this would stop the data collection every time we changed tabs. If the user wants to stop the collector, they could go back to the Statistics tab, remount the component (which should get the same performance collector of the scene), and stop using the button.